### PR TITLE
Enable the migration of 3 wmma API

### PR DESCRIPTION
**Pull request by wenhuiNi**
wmma API : `nvcuda::wmma::load_matrix_sync`, `nvcuda::wmma::mma_sync`, `nvcuda::wmma::store_matrix_sync`
wmma type : `nvcuda::wmma::mem_row_major`

### CODE REVIEW

File: clang/lib/DPCT/WMMAAPIMigration.cpp [hunk 1]
```diff
@@ -35,7 +42,11 @@ void clang::dpct::WMMARule::runRule(
     EA.analyze(*TL);
   } else if (const CallExpr *CE = getNodeAsType<CallExpr>(Result, "call")) {
     EA.analyze(CE);
-  } else {
+  } else if (const DeclRefExpr *DRE =
+                 getNodeAsType<DeclRefExpr>(Result, "enum")) {
+    DRE->dump();
```
Comment (yihwang-nv) [inline clang/lib/DPCT/WMMAAPIMigration.cpp]: Let's remove this debug dump

File: clang/lib/DPCT/MapNames.cpp [hunk 2]
```diff
@@ -1272,6 +1272,9 @@ void MapNames::setExplicitNamespaceMap() {
       {"CUSOLVER_EIG_RANGE_ALL", std::make_shared<EnumNameRule>("oneapi::mkl::rangev::all")},
       {"CUSOLVER_EIG_RANGE_V", std::make_shared<EnumNameRule>("oneapi::mkl::rangev::values")},
       {"CUSOLVER_EIG_RANGE_I", std::make_shared<EnumNameRule>("oneapi::mkl::rangev::indices")},
+      {"nvcuda::wmma::mem_row_major", 
+        std::make_shared<EnumNameRule>(
+            getClNamespace() + "ext::oneapi::experimental::matrix::layout::row_major")},
```
Comment (yihwang-nv) [inline clang/lib/DPCT/MapNames.cpp]: Do we need add a new option? this seems is a extension

File: clang/lib/DPCT/WMMAAPIMigration.cpp [hunk 3]
```diff
@@ -35,7 +42,11 @@ void clang::dpct::WMMARule::runRule(
     EA.analyze(*TL);
   } else if (const CallExpr *CE = getNodeAsType<CallExpr>(Result, "call")) {
     EA.analyze(CE);
-  } else {
+  } else if (const DeclRefExpr *DRE =
+                 getNodeAsType<DeclRefExpr>(Result, "enum")) {
+    DRE->dump();
+    EA.analyze(DRE);
+  }else {
```
Comment (yihwang-nv) [inline clang/lib/DPCT/WMMAAPIMigration.cpp]: Let's format the commit with `git-clang-format HEAD~1`

File: clang/lib/DPCT/ExprAnalysis.cpp [hunk 4]
```diff
@@ -491,7 +491,6 @@ void ExprAnalysis::analyzeExpr(const DeclRefExpr *DRE) {
         NameStringOS << ECD->getNameAsString();
         RefString = NameString;
       }
-
```
Comment (tangjj11) [inline clang/lib/DPCT/ExprAnalysis.cpp]: why modify this file

**Review by tangjj11 — APPROVED** [review]

**Review by yihwang-nv — CHANGES_REQUESTED** [review]
Basically LGTM % comments

**Review by yihwang-nv — APPROVED** [review]
LGTM

File: clang/lib/DPCT/CallExprRewriterWmma.cpp [hunk 5]
```diff
@@ -19,6 +21,13 @@ std::function<bool(const CallExpr *)> checkEnableJointMatrix() {
 }
 
 void CallExprRewriterFactoryBase::initRewriterMapWmma() {
+  if (DpctGlobalInfo::useExtJointMatrix()) {
```
Comment (AndyCHHuang) [inline clang/lib/DPCT/CallExprRewriterWmma.cpp:28]: Add a comment to explain which funciton migration requires the enum and why we cannot init it when construct EnumNamesMap?

**Review by AndyCHHuang — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]

**Pull request merged**